### ISSUE TITLE
GeoJSON provider doesn't implement `id` at root level

### ISSUE THREAD

**Issue by jorgejesus** [geopython/pygeoapi#499]
**Description**
GeoJSON provider doesn't implement `id` at root level. pygeoapi has 2 geojson:

https://github.com/geopython/pygeoapi/blob/master/tests/data/ne_110m_lakes.geojson
https://github.com/geopython/pygeoapi/blob/master/tests/data/ne_110m_populated_places_simple.geojson

The `id` is only present in `properties` content.

If `id` is removed from `properties` the data provider breaks

**Steps to Reproduce**

Remove id from properties and add it to the root of geojson 

**Expected behavior**

Expected id at root level and not in properties 
```
{
  "type": "FeatureCollection",
        "id": 0,
  "features": [
    {
      "type": "Feature",
      "properties": {
        "scalerank": 0,
        "name": "Lake Baikal",
        "name_alt": "https://en.wikipedia.org/wiki/Lake_Baikal",
        "admin": null,
        "featureclass": "Lake"
      },
      "geometry": {
        "type": "Polygon",
        "coordinates": []
```

**Screenshots/Tracebacks**

**Environment**
 - OS:
Linux
 - Python version:
 python 3.8.3
 - pygeoapi version:
master 

**Additional context**

**Comment by jorgejesus** [comment]
Closed by PR #501

**Comment by tomkralidis** [comment]
We need to be able to support an `id` either as `Feature` member or as a user specific value within `properties` (to support either case).

**Comment by alex-mathew** [comment]
Does the same apply to position of id field in GET response and POST payload?
Also what if there are 2 id fields - one as a feature member and other inside properties of the feature?

**Comment by jorgejesus** [comment]
IMHO, the `id` as a GeoJSON as a feature member should be the way to implement it and if not present then to check on `properties`.

**Comment by tomkralidis** [comment]
Any update on this one?

**Comment by alex-mathew** [comment]
@tomkralidis I have moved id field back to properties in sample lakes geojson file and have updated get method of geojson provider to first look for id field in feature level and if not found then in properties. Please review.

**Comment by tomkralidis** [comment]
Merged #515
